### PR TITLE
Various python dependencies for stable-diffusion-webui

### PR DESCRIPTION
**Pull request by mweinelt**
## Description of changes

Various python dependencies for stable-diffusion-webui.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/basicsr/default.nix [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  pname = "basicsr";
+  version = "1.4.2";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "XPixelGroup";
+    repo = "BasicSR";
+    rev = "v${version}";
+    hash = "sha256-UfwwwF0h0c5oPeGhj0w5Zw2edjPNoQWNG4pKHBwMU2I=";
+  };
```
Comment (SomeoneSerge) [inline pkgs/development/python-modules/basicsr/default.nix:37]: `CodeFormer` vendors a copy which contains some extra files. They're going to collide and I suspect we might want to just keep the copy?

Comment (mweinelt) [reply pkgs/development/python-modules/basicsr/default.nix:37]: Where do these extra files originate from? Really a fan of unvendoring this garbage as much as possible.

**Comment by mweinelt** [comment]
Planning to merge these packages early after branch-off, so they don't rot.

**Comment by SomeoneSerge** [comment]
I think if we introduce these packages we also have to package the various scripts that come with:

- https://github.com/xinntao/facexlib/tree/master/inference
- https://github.com/mlfoundations/open_clip/blob/main/scripts/clipav1_vit_l16_i37_t8.sh
- https://github.com/TencentARC/GFPGAN/blob/master/inference_gfpgan.py

These packages are the minority that do come with a `setup.py` (although probably still not actually intended for redistribution, seeing how e.g. they do not handle the entry points), and even have reasonable top-level names. If we keep adding dependencies, we're going to end up with many packages that 1) can only work with `catchConflicts = false`, 2) require writing our own `pyproject.toml`s, 3) come with scripts and "assets" expected at locations relative to `__file__`

Maybe we need to already choose a policy for handling these issues?

File: pkgs/development/python-modules/basicsr/default.nix (hunk 1 above)
Comment (SomeoneSerge) [reply pkgs/development/python-modules/basicsr/default.nix:37]: H'm maybe the vendored copy is just older, not sure if they actually patched it: https://gist.github.com/SomeoneSerge/595cf65e6acb86966e01b9f71363d3cd

**Pull request closed without merging**